packages/angular_devkit/core/node/host.ts

```diff
@@ -0,0 +1,320 @@
+              } else {
+                allFiles.push(normalize(path + '/' + fragment));
+              }
+            }
+          };
+          _recurseList(path);
+
+          return FromObservable.create(allFiles)
+            .pipe(
+              mergeMap(p => this._callFs(fs.unlink, p)),
+              ignoreElements(),
+              concat(FromObservable.create(allDirs).pipe(
+                concatMap(p => this._callFs(fs.rmdir, p)),
+              )),
+              map(() => {}),
+            );
+        } else {
+          return this._callFs(fs.unlink, path);
+        }
+      }),
+    );
+  }
+
+  rename(from: Path, to: Path): Observable<void> {
+    return this._callFs(fs.rename, from, to);
+  }
+
+  list(path: Path): Observable<PathFragment[]> {
+    return this._callFs(fs.readdir, path).pipe(
+      map(names => names.map(name => fragment(name))),
+    );
+  }
+
+  exists(path: Path): Observable<boolean> {
+    // Exists is a special case because it cannot error.
+    return new Observable(obs => {
+      fs.exists(path, exists => {
+        obs.next(exists);
+        obs.complete();
+      });
+    });
+  }
+
+  isDirectory(path: Path): Observable<boolean> {
+    return this._callFs(fs.stat, path).pipe(
+      map(stat => stat.isDirectory()),
+    );
+  }
+  isFile(path: Path): Observable<boolean> {
+    return this._callFs(fs.stat, path).pipe(
+      map(stat => stat.isDirectory()),
+    );
+  }
+
+  // Some hosts may not support stats.
+  stats(path: Path): Observable<virtualFs.Stats<fs.Stats>> | null {
+    return this._callFs(fs.stat, path);
+  }
+
+  // Some hosts may not support watching.
+  watch(
+    path: Path,
+    _options?: virtualFs.HostWatchOptions,
+  ): Observable<virtualFs.HostWatchEvent> | null {
+    return new Observable<virtualFs.HostWatchEvent>(obs => {
+      const watcher = new FSWatcher({ persistent: true }).add(path);
+
+      watcher
+        .on('change', path => {
+          obs.next({
+            path: normalize(path),
+            time: new Date(),
+            type: virtualFs.HostWatchEventType.Changed,
+          });
+        })
+        .on('add', path => {
+          obs.next({
+            path: normalize(path),
+            time: new Date(),
+            type: virtualFs.HostWatchEventType.Created,
+          });
+        })
+        .on('unlink', path => {
+          obs.next({
+            path: normalize(path),
+            time: new Date(),
+            type: virtualFs.HostWatchEventType.Deleted,
+          });
+        });
+
+      return () => watcher.close();
+    }).pipe(
+      publish(),
+      refCount(),
+    );
+  }
+}
+
+
+/**
+ * An implementation of the Virtual FS using Node as the backend, synchronously.
+ */
+export class NodeJsSyncHost implements virtualFs.Host<fs.Stats> {
+  get capabilities(): virtualFs.HostCapabilities {
+    return { synchronous: true };
+  }
+
+  write(path: Path, content: virtualFs.FileBuffer): Observable<void> {
```

Comment:
Feel a bit weird to have the sync host always return observables. Shouldn't it return what the observable is wrapping?

Reply:
Is it meant to be used with `SyncDelegateHost` maybe?

Reply:
Yes, it would be meant to be used with `SyncDelegateHost` if needed, but follow the `Host` interface so that it can be used where you need it.